src/TabViewTransitioner.js

```diff
@@ -158,20 +159,25 @@ export default class TabViewTransitioner extends Component<DefaultProps, Props,
   }
 
   _jumpToIndex = (index: number) => {
+    const { canJumpToTab, navigationState, onRequestChangeTab } = this.props;
+
+    if (canJumpToTab && !canJumpToTab(navigationState.routes[index])) {
+      return;
+    }
     this._triggerEvent('jump', index);
     this._nextIndex = index;
     this._transitionTo(index, () =>
       global.requestAnimationFrame(() => {
-        if (this.props.navigationState.index === index) {
```

Comment:
Props might have changed when this callback fires. So these shouldn't be changed, otherwise they'll be using old values.